### PR TITLE
Match css rewrite

### LINKED ISSUE #85: CSS logical properties syntax highlghting
Hi, just to mention that some of the newer css properties don't seem to be recognised by the Sass package. I'm thinking logical properties and some of the recent grid properties (e.g. place-items.) I've attached an image showing the issue. If you need more details just let me know.
Thanks
Screenshot 2021-08-12 at 19 38 05

### PR DESCRIPTION
**Pull request by braver**
The default CSS package was rewritten to solve various issues and align more closely with the newest scope naming guidelines. The changes are too big to "cherry pick", so we need to start over again as well. This PR tracks the process of starting from a fresh copy of the CSS package and then re-implementing the Sass language features on top of it. In the process we also simplify it around the `sass-value-expression`. 

----

Fixes #85 #82 #72 #69 #81

### CODE REVIEW

**Comment by jrappen** [comment]
Update the workflow? Compare:

<https://github.com/sublimehq/Packages/blob/master/.github/workflows/ci.yml>

**Comment by braver** [comment]
Oh, that can be updated as well, but for now it _is_ working... there are just quite a few tests that need looking at.

**Comment by jrappen** [comment]
Fair enough, though as a quick fix you might want to test against 4094 for the time being, as that was the last Build with syntax related changes.

File: Syntaxes/SCSS.sublime-syntax [hunk 1]
```diff
@@ -7,379 +7,3623 @@ file_extensions:
...
+  combinators: (?:>{1,3}|[~+]|\|{2})
+
+  vendor_prefix: -(?:webkit|moz|ms|o)-
+
+  # Custom Element Names
+  # https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-core-concepts
+  # adjusted from CSS so it doesn't match on interpolation (#{)
+  custom_element_tags: \b[a-z]{{custom_element_chars}}*-{{custom_element_chars}}*(?!#{){{break}}
+  # Note: Excludes `.` as it is used to identify attribute access
+  custom_element_chars: |-
```
Comment (jrappen) [inline Syntaxes/SCSS.sublime-syntax:78]: You might be able to use <https://github.com/kkos/oniguruma/blob/master/doc/UNICODE_PROPERTIES> via `\p{PROPERY_NAME}` for some of these.

**Comment by jrappen** [comment]
Was there a reason for **not using** [Inheritance](https://www.sublimetext.com/docs/syntax.html#inheritance:ver-dev)?

**Comment by braver** [comment]
> Was there a reason for not using Inheritance?

From Discord:

There are so many places I need to hook into to add support for expressions it becomes really hard to use inheritance. But more importantly, I cannot follow the Sublime release schedule because I simply don't have time. So whenever ST changes the CSS syntax that would potentially break LESS, SASS and SCSS. ST4 definitely would've broken those syntaxes. Since I can't pin my dependencies on a specific version, I cannot use dependencies.

**Comment by jrappen** [comment]
@braver The issue with d51e5e0 was: It should be `*.tar.xz` for the test binary, not `*.tar.bz2`. It changed around 4078 or so.

I see you're already working on it...

**Comment by jrappen** [comment]
@braver The Packages workflow expects subfolders. See comment above or compare with the one you replaced.

**Comment by braver** [comment]
I don't see the difference with what I had before, but now it's not seeing the yaml syntax I depend on. I'm too tired for this....

**Comment by jrappen** [comment]
I'll take another look maybe later tonight.

**Comment by braver** [comment]
Thanks, really appreciate it 👍🏻

**Comment by jrappen** [comment]
@braver I'd assume you'll need to pull in the Packages repo, if you depend on YAML.

**Comment by jrappen** [comment]
Randy rewrote the [UnitTesting](https://github.com/SublimeText/UnitTesting) plugin. Have a look there? I believe you can use stuff from there.

Or the usual suspect, [PackageDev](https://github.com/SublimeText/PackageDev) with [`syntax-test-action`](https://github.com/SublimeText/syntax-test-action).

**Comment by braver** [comment]
I guess it is seeing and interpreting my syntax correctly. In previous versions I didn’t have to do anything for the yaml stuff, but pulling in that syntax seems reasonable.

File: Syntaxes/SCSS.sublime-syntax [hunk 2]
```diff
@@ -7,379 +7,3624 @@ file_extensions:
...
-    - include: Sass.sublime-syntax#comment-line
-    - include: Sass.sublime-syntax#selectors
-    - include: properties
-    - include: Sass.sublime-syntax#keyframe-operators
-    - include: Sass.sublime-syntax#interpolated-selectors
-    - include: Sass.sublime-syntax#custom-element-selectors
+    # Yekyll front matter
+    - match: '^---$'
+      scope: frontmatter.jekyll punctuation.section.frontmatter.begin.jekyll
+      embed: scope:source.yaml
```
Comment (jrappen) [inline Syntaxes/SCSS.sublime-syntax:660]: (1) because of this

File: Syntaxes/Sass.sublime-syntax [hunk 3]
```diff
@@ -88,104 +650,15 @@ variables:
...
-    - match: '\s*(,)\s*'
-      captures:
-        1: punctuation.separator.css
-
 
-  frontmatter:
+    # Yekyll front matter
     - match: '^---$'
       scope: frontmatter.jekyll punctuation.section.frontmatter.begin.jekyll
       embed: scope:source.yaml
```
Comment (jrappen) [inline Syntaxes/Sass.sublime-syntax:664]: (2) and this

**Comment by jrappen** [comment]
Quickest fix should be using [`syntax-test-action`](https://github.com/SublimeText/syntax-test-action) with:

image

**Comment by braver** [comment]
Exactly, but before this PR is also had exactly that and the workflow did work then. Did previous versions of what it’s using from ST contain packages?

**Comment by jrappen** [comment]
Maybe you're never hitting that unchanged line? Just a random guess. Didn't have time to investigate.

**Comment by braver** [comment]
Could be. Thanks for all the tips and examples. I’ll figure it out after I get some sleep 😅

**Comment by niksy** [comment]
> > Was there a reason for not using Inheritance?
> 
> From Discord:
> 
> There are so many places I need to hook into to add support for expressions it becomes really hard to use inheritance. But more importantly, I cannot follow the Sublime release schedule because I simply don't have time. So whenever ST changes the CSS syntax that would potentially break LESS, SASS and SCSS. ST4 definitely would've broken those syntaxes. Since I can't pin my dependencies on a specific version, I cannot use dependencies.

Is this something that can be communicated with Sublime Text team so they can add basic support where we can hook in without rewriting everything? Maybe that’s something which can be useful to other syntaxes.

**Comment by jrappen** [comment]
FYI for ST4081+ there's also `view.syntax()` and [`sublime.Syntax`](https://www.sublimetext.com/docs/api_reference.html#sublime.Syntax) as an alternative to using `view.match_selector`.

**Comment by braver** [comment]
Progress: updated the completions file, and improved how property value completions work in SCSS. I notice the meta scopes in Sass could be better though, and I really need some time to look into this.

> where we can hook in without rewriting everything

There is nothing they can do to help speed up the development of this syntax. It really doesn't matter, in practice, if I copy-paste some of the CSS syntax, or if I can inherit them. I'm not rewriting their bits and pieces. Sure, it would be cleaner if I could depend on their work, but that's simply in theory impossible to do in a way that ensures they don't break my work.

The effort here in this branch isn't about copy-pasting stuff, it's making a syntax that is structurally different match up with an existing syntax. You'd think the nesting nature of SCSS is a minor thing, but it completely changes the game because anything can be a selector or a property and you can only be sure in retrospect. This is especially challenging in completions. You can't use meta scopes the same way to limit what can be matched, which means various bits I borrowed from the CSS syntax also need to be tweaked. 

I the end, it's just a lot of work, and if then the workflow also breaks, not all of it is actually any fun.

File: Syntaxes/SCSS.sublime-syntax (hunk 2 above)
Comment (braver) [reply Syntaxes/SCSS.sublime-syntax:660]: oh, sure, but apparently the 4073 build I was testing against doesn't fail over this but the 4121 build does 🤷🏻

**Comment by braver** [comment]
woop, passing tests now 💪🏻

**Pull request merged**